### ISSUE TITLE
Divide by zero in DateTimePrecise

### ISSUE THREAD

**Issue by igorberger** [fluentcassandra/fluentcassandra#48]
I'm seeing a divide by zero exception in FluentCassandra 1.1.3 with the following call stack:

System.DivideByZeroException: Attempted to divide by zero.
   at System.DateTimePrecise.GetUtcNow() in C:\FluentCassandra\V1.1.3\src\System\DateTimePrecise.cs:line 82
   at FluentCassandra.FluentColumn..ctor(CassandraColumnSchema schema) in C:\FluentCassandra\V1.1.3\src\FluentColumn.cs:line 33
   at FluentCassandra.FluentColumnFamily.TrySetColumn(Object name, Object value) in C:\FluentCassandra\V1.1.3\src\FluentColumnFamily.cs:line 235

This points to the following line in DateTimePrecise.GetUtcNow():

```
    DateTime tBaseNew = immutable.BaseTime.AddTicks(((
                    stopWatchTicks - immutable.ObservedTicks) * ClockTickFrequency) / (
                        immutable.StopWatchFrequency));
```

I can see only 2 places where an instance of DateTimePreciseSafeImmutable is created:
1. In the constructor of DateTimePrecise, where Stopwatch.Frequency is assigned to StopWatchFrequency.
   I've written a tiny test program to dump Stopwatch.Frequency to the console and it's not zero on the same box.
2. At the end of DateTimePrecise.GetUtcNow(), which has this in the denominator:

```
    utc.Ticks - immutable.ObservedTime.Ticks + utc.Ticks + utc.Ticks - tBaseNew.Ticks - immutable.ObservedTime.Ticks
```

When does this become zero?

P.S. If this expression is correct, then barring overflow issues, it can be simplified to:

```
    3 * utc.Ticks - 2 * immutable.ObservedTime.Ticks - tBaseNew.Ticks
```

**Comment by nberardi** [comment]
It seems to be a rare condition where the frequency is set exactly to zero by the 2nd condition that you identified.  To be honest, I didn't write this code and it was sort of hard to follow what was going on.  But I added a Math.Max(1, *) in the immutable object to make sure it is never zero.

**Comment by igorberger** [comment]
Thanks for the fix. When do you think you'll tag it with a new verison?

**Comment by nberardi** [comment]
I have a couple more pull requests to merge in then I will put out a point
release.

**Comment by igorberger** [comment]
I'm using 1.1.4 now and I noticed the following in the logs, which looks like an aftershock of this change:

System.ArgumentOutOfRangeException: The added or subtracted value results in an un-representable DateTime.
Parameter name: value
   at System.DateTime.AddTicks(Int64 value)
   at System.DateTimePrecise.GetUtcNow() in C:\FluentCassandra\V1.1.4\src\System\DateTimePrecise.cs:line 81
   at FluentCassandra.FluentColumn..ctor(CassandraColumnSchema schema) in C:\FluentCassandra\V1.1.4\src\FluentColumn.cs:line 36
   at FluentCassandra.Operations.Helper.ConvertColumnToFluentColumn(Column col, CassandraColumnSchema colSchema) in C:\FluentCassandra\V1.1.4\src\Operations\Helper.cs:line 205
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at FluentCassandra.FluentColumnList`1..ctor(FluentColumnParent parent, IEnumerable`1 columns) in C:\FluentCassandra\V1.1.4\src\FluentColumnList`1.cs:line 35
   at FluentCassandra.FluentColumnFamily..ctor(CassandraObject key, String columnFamily, CassandraColumnFamilySchema schema, IEnumerable`1 columns) in C:\FluentCassandra\V1.1.4\src\FluentColumnFamily.cs:line 54
   at FluentCassandra.Operations.MultiGetColumnFamilySlice.<Execute>d__5.MoveNext() in C:\FluentCassandra\V1.1.4\src\Operations\MultiGetColumnFamilySlice.cs:line 37
   at System.Linq.Enumerable.SingleOrDefault[TSource](IEnumerable`1 source)

**Comment by nberardi** [comment]
I think ultimately, I am going to have to remove this DateTimePrecise.  It gives a much more accurate resolution on the timestamp, but it is troublesome.

**Comment by igorberger** [comment]
Thanks for keeping this issue open.

Is there something we can do in the meantime to avoid this problem, since it's actually affecting us?

**Comment by nberardi** [comment]
Let me put out an emergency patch for you via NuGet it will be 1.1.6.

**Comment by nberardi** [comment]
DateTimePrecise has been removed, I will be looking for ways to increase the resolution of the Standard DateTimeOffset object, but for right now we are just going to have to deal with it's resolution.

**Comment by igorberger** [comment]
Thank you for a very quick fix.

**Comment by edweip3** [comment]
Thanks! I was just about to comment on this issue..  the real problem with this class is that as time goes on, you get arithmetic overflow when doing the calculation, causing the division by zero problem.  setting it to be 1 when it does become zero causes timestamp problem experience by @igorberger.  Glad this is removed for now.
